lib/components/Editor/utils.js

```diff
@@ -39,3 +39,12 @@ export const setInitialPosition = editor => {
   transaction.setSelection(endPosition);
   view.dispatch(transaction);
 };
+
+export const isEditorTippyContentHidden = () =>
+  !document.getElementsByClassName("tippy-content")?.length;
+
+export const isEditorImageModalHidden = () =>
+  !document.getElementsByClassName("ne-media-uploader")?.length;
+
+export const isEditorOverlaysActive = () =>
+  isEditorTippyContentHidden() && isEditorImageModalHidden();
```

Comment:
If I remember correctly, I had pointed out the exact changes that were over the call. Why did you just directly copy paste the method from neetoPlanner?

This is how it needs to look:

```jsx
export const isEditorOverlaysActive = (ref) => {
  const isTippyContentActive = ...;
  const isOverlaysActive = ...;

  return isTippyContentActive || isOverlaysActive;
}
```

1. We don't need three methods, move two of the methods as variables.
2. We need not check the whole document. Check only on the reference element that is provided.
3. `isEditorImageModalHidden` should also check for all the modals that can be opened by the editor. Embeds are also opening a modal.

LMK if any more clarifications are needed.

Reply:
> 2\. We need not check the whole document. Check only on the reference element that is provided.

@AbhayVAshokan I tried passing in the editorWraperRef but the ref.current.getElementsByClassName("ne-media-uploader") returns empty array that is why I opted to stick with the same code in neetoPlanner.

Reply:
Aah yes, the modal will be within a React Portal. That is why `ref.current` cannot track it. Please make the other changes.